demo/demo.html

```diff
@@ -4,29 +4,72 @@
+
+    <sfx-search-box id="secondary-search" group="secondary" placeholder="search" clearbutton searchbutton></sfx-search-box>
+    <sfx-sayt searchbox="secondary-search" group="secondary" visible></sfx-sayt>
+
+    <sfx-products></sfx-products>
 
     <script>
+      function parseRecord(record) {
+        const data = record.allMeta;
+        const firstVariant = data.visualVariants[0];
+        const nonvisualVariants = firstVariant.nonvisualVariants;
+        if (!nonvisualVariants[0]) throw new Error('No nonvisual variants');
+
+        return {
+          data,
+          firstVariant,
+          nonvisualVariants,
+        };
+      }
+
+      function productTransformer(record) {
+        let filter;
+        try {
+          filter = parseRecord(record);
+        } catch(error) {
+          console.error(error);
+          return;
+        }
+        const { data, firstVariant, nonvisualVariants } = filter;
+
+        return {
+          title: data.title,
+          price: nonvisualVariants[0].originalPrice,
+          imageSrc: firstVariant.productImage,
+          imageAlt: data.title,
+          productUrl: firstVariant.productImage,
+          // @TODO Handle variants
+        }
+      }
+
+      function searchProductTransformer(record) {
+        return { label: record.title };
+      }
+
       // init core and plugins
       const core = new sfxCore.Core();
       const domEventsPlugin = new sfxPlugins.DomEventsPlugin()
-      const saytPlugin = new sfxPlugins.SaytPlugin({subdomain: 'apparel', collection: 'products'});
-      const saytDriverPlugin = new sfxPlugins.SaytDriverPlugin();
-      const searchPlugin = new sfxPlugins.SearchPlugin({ customerId: 'apparel', collection: 'products', area: 'Storefront'});
-      const searchDriverPlugin = new sfxPlugins.SearchDriverPlugin();
+      const saytPlugin = new sfxPlugins.SaytPlugin({ subdomain: 'apparel' });
+      const saytDriverPlugin = new sfxPlugins.SaytDriverPlugin({ productTransformer });
+      const searchPlugin = new sfxPlugins.SearchPlugin({ customerId: 'apparel' });
+      const searchDriverPlugin = new sfxPlugins.SearchDriverPlugin({ productTransformer: searchProductTransformer });
       // register all plugins with core
       core.register([saytPlugin, saytDriverPlugin, domEventsPlugin, searchPlugin, searchDriverPlugin]);
-      // search integration/functionality test
-      window.addEventListener('sfx::search_response', (e) => {
-        const testListResults = e.detail.results.records.map(item => {
-          return { label : item.title };
-        });
-        document.getElementsByClassName('product-grid')[0]
-        .setAttribute('items', JSON.stringify(testListResults));
+
+      window.addEventListener('sfx::sayt_products_response', (event) => {
+        console.log(event.type, event.detail);
+      });
+
+      window.addEventListener('sfx::sayt_products_error', (event) => {
+        console.log(event.type, event.detail);
       });
```

Comment:
I can remove these on request, but it seems reasonable (and time saving) to have a bit of logging in our demo.

Reply:
We should consider having listeners for all our events. Maybe later.

Plugin idea: we have a plugin that depends on a Logger plugin (yet to be built) and the Events plugin. It would subscribe to a list of events and log out its name and payload upon receipt.

Reply:
I do like this idea. We could instantiate it on our demo and show clients that they can always use it to understand how SF-X is hooking up.